torch/csrc/jit/attributes.h

```diff
@@ -19,10 +19,21 @@ using ::c10::Symbol;
 
 constexpr int max_tensor_display_size = 10;
 
-enum class AttributeKind { f, fs, i, is, s, ss, t, ts, g, gs, ty, tys };
+enum class AttributeKind { f, fs, i, is, s, ss, t, ts, g, gs, ty, tys, tup };
```

Comment:
I have concerns about the gradient of this change. Every additional value that needs to be a constant will now need to be re-encoded as a attribute in the ir. As this expands, IR will reproduce the functionality of IValue. If we were starting from scratch, the _only_ attribute type a Node would have would be an IValue. We can't easily do that now, because too many downstream things, like onnx, use attributes. However, I think it is reasonable to add IValue as an attribute type and use it for tuple constants. The way I see this going is:

1. This Patch: adds IValue as an attribute type, along with support for using it to hold Tuple constant. Should be a small change to this patch because handling of other constants remain the same.
2. IValues gain a repr (i.e. how do I serialize this thing as python text, so that python_print can read it). Switch uses of quoted strings to use repr as well and clean up python_print.
3. Store  f, fs, i, is, s, ss as IValues as well. So that we do not change the externally visible API, the f, f_ functions should remain, but they should just do a type check that the underlying IValue is a float (or int, or string). Constants should now work by just directly storing the IValue.

Reply:
So, as far as this patch goes, what is the concrete difference ? instead of `TupleAttr` there is `IValueAttr`, and the `tup_` accessor converts the Tuple IValue into the general IValueAttr ?

As far as repr: that seems like a good change, containers such as tuple will still need to be special cased for python print because of the special `CONSTANTS.0` exporting of tuple constants